### PR TITLE
Idea for extract_levels check

### PR DESCRIPTION
**Pull request by bouweandela**
This adds an extra check on extract_levels settings

Before you start, please read [CONTRIBUTING.md](https://github.com/ESMValGroup/ESMValTool/blob/version2_development/CONTRIBUTING.md).

**Tasks**

-   [ ] [Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do, if you haven't done so already (and add the link at the bottom)
-   [ ] Add unit tests
-   [ ] Public functions should have a numpy-style docstring so they appear properly in the [API documentation](https://esmvaltool.readthedocs.io/projects/esmvalcore/en/latest/api/esmvalcore.html). For all other functions a one line docstring is sufficient.
-   [ ] If writing a new/modified preprocessor function, please update the [documentation](https://esmvaltool.readthedocs.io/projects/esmvalcore/en/latest/esmvalcore/preprocessor.html)
-   [ ] Circle/CI tests pass. Status can be seen below your pull request. If the tests are failing, click the link to find out why.
-   [ ] Codacy code quality checks pass. Status can be seen below your pull request. If there is an error, click the link to find out why. If you suspect Codacy may be wrong, please ask by commenting.
-   [ ] Please use `yamllint` to check that your YAML files do not contain mistakes
-   [ ] If you make backward incompatible changes to the recipe format, make a new pull request in the [ESMValTool repository](https://github.com/ESMValGroup/ESMValTool) and add the link below

If you need help with any of the tasks above, please do not hesitate to ask by commenting in the issue or pull request.

* * *

Closes {Link to corresponding issue}

### CODE REVIEW

File: esmvalcore/_recipe_checks.py [hunk 1]
```diff
@@ -138,3 +139,19 @@ def extract_shape(settings):
...
+def extract_levels(settings, variable):
+    short_name = variable['short_name']
+    table_entry = CMOR_TABLES[variable['cmor_table']].get_variable(
+        variable['mip'],
+        short_name,
+        variable.get('derive', False),
+    )
+    for coordinate in table_entry.coordinates.values():
+        if coordinate.axis == 'Z':
+            break
```
Comment (valeriupredoi) [inline esmvalcore/_recipe_checks.py]: careful here with generic coordinates like `olevel` or `alevel` -> are they mapped to eg coord name `plev` at this stage? Can't remember @jvegasbsc

**Comment by schlunma** [comment]
@bouweandela do you have any intentions to further work on this for v2.5?

**Comment by schlunma** [comment]
I just had another look at this PR and it seems that it's not necessary anymore. We already have a check for the existence of a `Z` coordinate here (l.1022):

https://github.com/ESMValGroup/ESMValCore/blob/7d430722af8a3aee206287edb210eedc87ca4fcb/esmvalcore/preprocessor/_regrid.py#L972-L1025

Please re-open if necessary.

**Comment by bouweandela** [comment]
The advantage of this check was that it does not need to open the file containing the data when parsing the recipe, it checks if the level is in the CMOR table instead. So I think this would still be useful (especially considering #430).

**Comment by schlunma** [comment]
Make sense. As I said, please re-open if necessary. Do you think this is something for v2.5?

**Pull request closed without merging**